doc/src/devdocs/sysimg.md

```diff
@@ -88,6 +88,24 @@ Additionally, a few special features are supported to control the function cloni
     This cause the function for the targe to be optimize for size that might have
     a significant runtime performance impact. This corresponds to `-Oz` Clang option.
 
+As an example, at the time of this writing, the following string is used in the creation of
+the official `x86_64` Julia binaries downloadable from julialang.org:
+
+```
+generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)
+```
+
+This creates a system image with three separate targets; one for a generic `x86_64`
+processor, one with a `sandybridge` ISA (explicitly excluding `xsaveopt`) that explicitly
+clones all functions, and one targeting the `haswell` ISA, based off of the `sandybridge`
+sysimg version, and also excluding `rdrnd`.  When a Julia implementation loads the
+generated sysimg, it will check the host processor for matching CPU capability flags,
+enabling the highest ISA level possible.  Note that the base level (`generic`) requires
+the `cx16` instruction, which is disabled in some virtualization software and must be
+enabled for the `generic` target to be loaded.  Alternatively, a sysimg could be generated
+with the target `generic,-cx16` for greater compatibility, however note that this may cause
```

Comment:
@yuyichao Do you recall why we require `cx16`

Reply:
Yes, I believe this is the still complicated issue of ABI compliance of 128bit atomics.

1. GCC behavior. It always emit a `__atomic_*` libcall.
2. Clang behavior. It uses `cmpxchg16b` instruction when available and `__sync_lock_test_and_set_16` if not.

There are arguments suggesting that the GCC one is the only correct behavior since the clang behavior does not allow atomic load on constant memory but it does mimic the current implementation in `libatomic` and it seems that they are not changing that any time soon...

For us, if we don't require `cx16`, we need a way to inject a `__sync_lock_test_and_set_16` implementation that makes `llvm` happy.

Reply:
Right so we would have #18706